pkg/manifests/manifests.go

```diff
@@ -1898,6 +1962,40 @@ func (f *Factory) setupQueryLogFile(p *monv1.Prometheus, queryLogFile string) er
 	return nil
 }
 
+func (f *Factory) setupScrapeProfiles(p *monv1.Prometheus, scrapeProfile string) error {
+	if scrapeProfile == "" {
+		scrapeProfile = "full"
+	}
+
+	if scrapeProfile != "full" && scrapeProfile != "operational" && scrapeProfile != "uponly" {
+		return errors.Wrap(ErrConfigValidation, `scrape profile provided is unknown, supported scrape profiles are: full, operational, uponly`)
+	}
```

Comment:
I'd do the verification and defaulting beforehand, e.g. in https://github.com/openshift/cluster-monitoring-operator/blob/274ceee6174dfa776db0a7f20c6e3a4100ca0f18/pkg/manifests/config.go#L189-L201

Reply:
I agree with the defaulting however, regarding validation we usually do it on `manifests/manifests.go` so I'm not so sure we would want to move it to `manifests/config.go`. But do let me know if you prefer it that way!

Reply:
I'm more in favor of bailing out early. It also means that ` NewConfig()` returns a semantically correct value and we don't have to worry about it later.